src/loaddb/load_session.cpp

```diff
@@ -289,8 +317,23 @@ namespace cubload
   {
     delete m_driver;
 
-    cubthread::get_manager ()->destroy_worker_pool (m_worker_pool);
-    delete m_wp_context_manager;
+    loaddb_wp_mutex.lock ();
```

Comment:
move block worker related code block to another function - load_wp_unregister_session .